jujugui/static/gui/src/app/views/utils.js

```diff
@@ -1810,6 +1810,57 @@ YUI.add('juju-view-utils', function(Y) {
     return multiple ? utils.numToLetter(multiple) + char : char;
   };
 
+  /**
+    Generate a name for a service with an incremented number if needed.
+
+    @method generateServiceName
+    @param {String} charmName The charm name.
+    @param {String} charmId the full charm id.
+    @param {Object} services the list of services.
+    @param {Boolean} ignoreSelf ignore this service in the service count.
+    @returns {String} The name for the service.
+  */
+  utils.generateServiceName = function(
+    charmName, charmId, services, ignoreSelf) {
+    var charmCount = 0;
+    // Loop through each service and grab its charmid to see if it matches.
+    services.each((service) => {
+      // Remove the version number from the charm id as we don't care about the
+      // version when comparing charms.
+      var serviceCharmId = utils._extractCharmName(service.get('charm'));
+      var providedCharmId = utils._extractCharmName(charmId);
+      // If we have a matching charm then increase the count.
+      if (serviceCharmId === providedCharmId) {
+        charmCount += 1;
+      }
+    });
+    // In some cases the service has already been added to the db so we need to
+    // ignore it in the final count.
+    if (ignoreSelf) {
```

Comment:
I don't really understand this - it's supposed to be generating the name to be added to the db...so how could it be in the db already?